src/migrate_parsetree_driver.ml

```diff
@@ -305,44 +305,41 @@ type ('a, 'b) intf_or_impl =
 
 let guess_file_kind fn =
   if Filename.check_suffix fn ".ml" then
-    Impl fn
+    `Impl
```

Comment:
Could you use a normal variant here? I'm not a big fan of polymorphic variant when normal ones work just as well

Reply:
I replaced it by a normal variant, but that introduces an `assert false` in one place. I guess that can be worked around using more transformations.